src/com/esotericsoftware/kryonet/TcpConnection.java

```diff
@@ -46,10 +46,14 @@
 	private volatile long lastWriteTime, lastReadTime;
 	private int currentObjectLength;
 	private final Object writeLock = new Object();
+	private final int objectBufferSize;
+	private final int lengthLength;
 
 	public TcpConnection (Serialization serialization, int writeBufferSize, int objectBufferSize) {
 		this.serialization = serialization;
-		writeBuffer = ByteBuffer.allocate(writeBufferSize);
+		this.objectBufferSize = objectBufferSize;
+		lengthLength = serialization.getLengthLength();
+		writeBuffer = ByteBuffer.allocate(Math.max(writeBufferSize, lengthLength + objectBufferSize));
```

Comment:
Why not use the write buffer size that was passed in?

Reply:
Thanks, I prefer an exception to using a buffer size other than what was specified.